doop.c

```diff
@@ -635,6 +635,37 @@ Perl_do_trans(pTHX_ SV *sv)
     }
 }
 
+static int
+S_sv_has_magic_method(pTHX_ SV *sv, int method)
+{
+    HV* stash = NULL;
+    MAGIC *mg = NULL;
+    CV **cvp = NULL;
+    CV *cv = NULL;
+
+    if (!SvAMAGIC(sv))
+        return 0;
+
+    stash = SvSTASH(SvRV(sv));
+    if (!Gv_AMG(stash))
+        return 0;
+
+    mg = mg_find((const SV *)stash, PERL_MAGIC_overload_table);
+    if (!mg)
+        return 0;
+
+    if (AMT_AMAGIC((AMT*)mg->mg_ptr))
+        cvp = ((AMT*)mg->mg_ptr)->table;
```

Comment:
your use of casts in this function is inconsistent. `(const SV *)` versus `(AMT*)` IMO the latter should be (AMT *).